src/layers/layerHelper.js

```diff
@@ -1,3 +1,29 @@
+const zoomEquivalent = {
+  7: 1,
+  8: 1,
+  9: 2,
+  10: 3,
+  11: 4,
+  12: 5,
+  13: 6,
+  14: 7,
+  15: 8,
+  16: 8,
+  17: 9,
+  18: 10,
+};
+
+// Allow to convert zoom from new wkp to old one.
+function convertToOldZoom(zoom) {
+  if (zoom < 9) {
+    return 1;
+  }
+  if (zoom > 18) {
+    return 10;
+  }
+  return zoomEquivalent[zoom];
+}
```

Comment:
@edellucien What about decimal numbers as zoom value?

Reply:
In the Permalink I ensure the use of an integer as zoom:
`parseInt(urlParams.z, 10)`
Or would you recommand to parse to Int in the convertToOldZoom func?

Reply:
Now, I see it. Yes, I think it is better if you include it in the function `convertToOldZoom()`.

Reply:
Short version suggestion with `parseInt()`:
`return zoomEquivalent[Math.min(Math.max(parseInt(zoom, 10), 9), 18)];`